src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,29 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+
+        public static string GetMeaningfulParentName(SyntaxNode current, SemanticModel model)
```

Comment:
It's not clear what's the purpose of this method from the name of it. Also consider move this to desktop analyzer helper.